chapters/untyped_pointers.md

```diff
@@ -0,0 +1,129 @@
+
+This idea can also be found in the LLVM community, which has moved to `Opaque Pointers` starting in LLVM 15.
+
+> More information can be found in https://github.com/KhronosGroup/Vulkan-Docs/blob/main/proposals/VK_KHR_shader_untyped_pointers.adoc
+
+# What the extension adds
+
+This extension adds various `Untyped` instructions, such as:
+
+- Op**Untyped**VariableKHR`
+- Op**Untyped**AccessChainKHR`
+- OpType**Untyped**PointerKHR`
+
+> `OpTypeUntypedPointerKHR` may seem strange, but it follows the `OpType*` convention and is just unfortunate the "type" and "untyped" are both in the name
+
+# Rethinking the pointer type
+
+The better understand this extension, first focus on `OpTypeUntypedPointerKHR` as that is the "core" change. By using this new pointer type, you will find the reason the other "untyped" instructions were added to go along with it.
+
+Let's use this very basic example of load and storing to a storage buffer ([view online](https://godbolt.org/z/5Wx1PsxG7)):
+
+```glsl
+layout(set = 0, binding = 0) buffer SSBO {
+    uint a;
+};
+
+void main() {
+    uint b = a;
+    a = 0;
+}
+```
+
+This will generate into something such as
+
+```swift
+   %uint = OpTypeInt 32 0
+ %uint_0 = OpConstant %uint 0
+
+    %SSBO = OpTypeStruct %uint
+%SSBO_ptr = OpTypePointer StorageBuffer %SSBO
+%variable = OpVariable %SSBO_ptr StorageBuffer
+
+ %ptr = OpTypePointer StorageBuffer %uint
+
+  %ac = OpAccessChain %ptr %variable %uint_0
+%load = OpLoad %uint %ac
+        OpStore %ac %uint_0
+```
+
+Notice that `OpLoad` and `OpStore` both have two operands, and **both** can figure out the type is `%uint`. This is the **core motivation** of this extension, to remove the need for having to declare the `%uint` type here in the `Pointer` operand.
+
+With untyped pointer, we still want will use the access chain to get the `StorageClass`, but we can now remove the redundancy by first turning the pointer into an untyped pointer.
+
+```diff
+   %uint = OpTypeInt 32 0
+ %uint_0 = OpConstant %uint 0
+
+- %ptr = OpTypePointer StorageBuffer %uint
++ %ptr = OpTypeUntypedPointerKHR StorageBuffer
+
+    %SSBO = OpTypeStruct %uint
+%SSBO_ptr = OpTypePointer StorageBuffer %SSBO
+%variable = OpVariable %SSBO_ptr StorageBuffer
+
+  %ac = OpAccessChain %ptr %variable %uint_0
+%load = OpLoad %uint %ac
+        OpStore %ac %uint_0
+```
+
+But this is **stil invalid**! We now need a matching untyped access chain as well where we now have a new `Base Type` operand.
+
+```diff
+   %uint = OpTypeInt 32 0
+ %uint_0 = OpConstant %uint 0
+
+ %ptr = OpTypeUntypedPointerKHR StorageBuffer
+
+    %SSBO = OpTypeStruct %uint
+%SSBO_ptr = OpTypePointer StorageBuffer %SSBO
+%variable = OpVariable %SSBO_ptr StorageBuffer
+
+-  %ac = OpAccessChain %ptr %variable %uint_0
++  %ac = OpUntypedAccessChainKHR %ptr %SSBO %variable %uint_0
```

Comment:
This is, in fact, unnecessary and might be worth calling out. This access chain calculates an offset of 0 from `%variable` and is equivalent to just using the variable itself. The `%uint_0` was necessary in typed pointers to satisfy type rules.